noir-projects/noir-contracts/contracts/app/crowdfunding_contract/src/main.nr

```diff
@@ -78,10 +83,12 @@ pub contract Crowdfunding {
         // contract by proving that the hash of this note exists in the note hash tree.
         let note = UintNote::new(amount, donor);
 
-        storage.donation_receipts.insert(note).emit(
+        // We don't constrain encryption because the donor is sending the note to himself. Hence by performing
+        // encryption incorrectly would harm himself only.
+        storage.donation_receipts.at(donor).insert(note).emit(
             &mut context,
             donor,
-            MessageDelivery.CONSTRAINED_ONCHAIN,
+            MessageDelivery.UNCONSTRAINED_ONCHAIN,
```

Comment:
Unrelated change. Using constrained encryption here was incorrect so I updated it.